### PR TITLE
Fixes multitile doors and multitile engines on shuttles. Gives raven it's big engines back

### LINKED ISSUE #24610: horizontal multi-tile atoms are not transited properly on shuttles
### Exploit Reports

- [X] I confirm this issue is not an exploit. (Required)

### BYOND Version

515.1619

### Issue Description

Doors that are 2x1 wide horizontal do not maintain their proper positioning on shuttles when moving between docking ports.

### What did you expect to happen?

These things should stay where they were placed.

### What happened instead?

They don't.

Initial map placement:
2024_03_13__17_55_40__paradise dme  whiteship dmm  - StrongDMM

In-game:
2024_03_13__17_51_02__Paradise Station 13

In-transit
2024_03_13__17_51_11__Paradise Station 13

After reaching next docking port
2024_03_13__17_51_17__Paradise Station 13

### Why is this bad/What are the consequences?

It prevents the ability to use these items when mapping.

### Steps to reproduce the issue.

Map wide airlocks onto a shuttle such as the white ship. Spawn in, and move the shuttle to another docking port.

### When did the problem start happening?

Who knows.

### Extra information

Ffffucking shuttle code

### Relevant log output/runtime error

_No response_

### PR DESCRIPTION
**Pull request by Qwertytoforty**
## What Does This PR Do

Fixes multitile doors and multitile engines on shuttles. Also pods if they existed still.
Grants the NTSS Raven it's large engines it was intended to have, but I removed when porting due to the bug that this fixes.
Fixes #24610

## Why It's Good For The Game

Bugs are bad.
More creative shuttles with huge engines good.
Raven having big boy engines is good.

## Testing

example on discord

## Changelog
:cl:
tweak: Gives NTSS Raven it's big engines back, gives whiteship multi tile cockpit per request
fix: Fixes multitile doors and multitile engines on shuttles.
/:cl:

### CODE REVIEW

File: code/modules/shuttle/shuttle_rotate.dm [hunk 1]
```diff
@@ -24,6 +24,19 @@ If ever any of these procs are useful for non-shuttles, rename it to proc/rotate
...
 			pixel_y = (oldPX*(-1))
 
+
+/atom/movable/shuttleRotate(rotation, params)
+	. = ..()
+	//rotate the physical bounds and offsets for multitile atoms too. Owerride base "rotate the pixel offsets" for multitile atoms.
+	//Owerride non zero bound_x, bound_y, pixel_x, pixel_y to zero.
+	//Dont take in account starting bound_x, bound_y, pixel_x, pixel_y.
+	//So it can unintentionally shift physical bounds of things that starts with non zero bound_x, bound_y.
+	if(((bound_height != world.icon_size) || (bound_width != world.icon_size)) && (bound_x == 0) && (bound_y == 0)) //Dont shift things that have non zero bound_x and bound_y, or it move somewhere. Now it BSA and Gateway.
```
Comment (1080pCat) [inline code/modules/shuttle/shuttle_rotate.dm:34]: What does the `Now it BSA and Gateway` mean?

File: code/modules/shuttle/shuttle_rotate.dm [hunk 2]
```diff
@@ -24,6 +24,19 @@ If ever any of these procs are useful for non-shuttles, rename it to proc/rotate
 			pixel_x = oldPY
 			pixel_y = (oldPX*(-1))
 
+
+/atom/movable/shuttleRotate(rotation, params)
+	. = ..()
+	//rotate the physical bounds and offsets for multitile atoms too. Owerride base "rotate the pixel offsets" for multitile atoms.
+	//Owerride non zero bound_x, bound_y, pixel_x, pixel_y to zero.
```
Comment (1080pCat) [inline code/modules/shuttle/shuttle_rotate.dm:31]:
```suggestion
	//Override non zero bound_x, bound_y, pixel_x, pixel_y to zero.
```

File: code/modules/shuttle/shuttle_rotate.dm [hunk 3]
```diff
@@ -24,6 +24,19 @@ If ever any of these procs are useful for non-shuttles, rename it to proc/rotate
 			pixel_x = oldPY
 			pixel_y = (oldPX*(-1))
 
+
+/atom/movable/shuttleRotate(rotation, params)
+	. = ..()
+	//rotate the physical bounds and offsets for multitile atoms too. Owerride base "rotate the pixel offsets" for multitile atoms.
```
Comment (1080pCat) [inline code/modules/shuttle/shuttle_rotate.dm:30]:
```suggestion
	//rotate the physical bounds and offsets for multitile atoms too. Override base "rotate the pixel offsets" for multitile atoms.
```

**Review by 1080pCat — COMMENTED** [review]
mapdif shows a lack of the big engines, they do seem to exist in sdmm though

File: code/modules/shuttle/shuttle_rotate.dm (hunk 1 above)
Comment (Qwertytoforty) [reply code/modules/shuttle/shuttle_rotate.dm:34]: Removing, those on tg are not rotated, iirc

File: code/modules/shuttle/shuttle.dm [hunk 4]
```diff
@@ -508,6 +508,8 @@
 
 			//move mobile to new location
 			for(var/atom/movable/AM in T0)
+				if(AM.loc != T0) //fix for multi-tile objects
```
Comment (Contrabang) [inline code/modules/shuttle/shuttle.dm:511]: Hacky, but it looks like it works given the logic

File: code/modules/shuttle/shuttle_rotate.dm [hunk 5]
```diff
@@ -24,6 +24,19 @@ If ever any of these procs are useful for non-shuttles, rename it to proc/rotate
...
+
+/atom/movable/shuttleRotate(rotation, params)
+	. = ..()
+	//rotate the physical bounds and offsets for multitile atoms too. Override base "rotate the pixel offsets" for multitile atoms.
+	//Override non zero bound_x, bound_y, pixel_x, pixel_y to zero.
+	//Dont take in account starting bound_x, bound_y, pixel_x, pixel_y.
+	//So it can unintentionally shift physical bounds of things that starts with non zero bound_x, bound_y.
+	if(((bound_height != world.icon_size) || (bound_width != world.icon_size)) && (bound_x == 0) && (bound_y == 0)) //Dont shift things that have non zero bound_x and bound_y, or it move somewhere.
+		pixel_x = dir & (NORTH|EAST) ? -bound_width+world.icon_size : 0
+		pixel_y = dir & (NORTH|WEST) ? -bound_width+world.icon_size : 0
```
Comment (Contrabang) [inline code/modules/shuttle/shuttle_rotate.dm:36]:
```suggestion
		pixel_x = dir & (NORTH|EAST) ? (world.icon_size - bound_width) : 0
		pixel_y = dir & (NORTH|WEST) ? (world.icon_size - bound_width) : 0
```

**Review by Contrabang — CHANGES_REQUESTED** [review]

**Review by Contrabang — CHANGES_REQUESTED** [review]

**Pull request merged**